T5713: Strip string after "secret" in IPSEC configs

sempervictus opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change Summary
Makes `strip-private` strip the string after "secret" to avoid:
```
# show | strip-private| match secret
                 secret someactualstring
```
## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
* T5713

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->

## Component(s) name
CLI

## Proposed changes
<!--- Describe your changes in detail -->

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
$ configure
# show | strip-private| match secret
```
verify the string after secret is replaced with `*`s
-->

## Smoketest result
<!-- Provide the output of the smoketest
```
$ /usr/libexec/vyos/tests/smoke/cli/test_xxx_feature.py
test_01_simple_options (__main__.TestFeature.test_01_simple_options) ... ok
```
-->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [ ] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [ ] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [ ] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

Commit:
T5713: Strip string after "secret" in IPSEC config

Make "strip-private" strip the string after "secret"

c-po commented on src/helpers/strip-private.py:
```diff
@@ -119,6 +119,7 @@ def strip_lines(rules: tuple) -> None:
         (True, re.compile(r'(shared-secret-key-file|ca-cert-file|cert-file|dh-file|key-file|client) (\S+)'), r'\1 xxxxxx'),
         # Strip IPSEC secrets
         (True, re.compile(r'pre-shared-secret \S+'), 'pre-shared-secret xxxxxx'),
+        (True, re.compile(r'\ssecret \S+'), ' secret xxxxxx'),
```
Why is a leading whitespace required and not only `secret`?

sempervictus replied:
Generic word which might later appear at the end of another string like `use-secret` so figured I'd capture whole

c-po replied:
Good intention but this will happen:

`use-secret foo` CLI node will become ` secret xxxxxx` so the output of `strip-private` invalidates the configuration.

Please just make it as simple as possible and use:

`(True, re.compile(r'secret \S+'), 'secret xxxxxx'),`

c-po reviewed (changes requested):
It might be an easy fix but it still requires a task id. Please create a bug report in Phabricator and add the Task ID to the commit message.

We require this for proper release tracking and release node generation/backporting

sever-sever commented:
Wrong format of commit message

sempervictus commented:
> Wrong format of commit message

Sorry, thought fixing the GH title might suffice. Reworded/rebased and force-pushed.

c-po commented:
Looks like your rebase re-added one of my latest commits


![image](https://github.com/vyos/vyos-1x/assets/25299219/e6f256ae-d315-4984-964d-4d746e170a2f)

sempervictus commented:
> Wrong format of commit message

Sorry, thought fixing the GH title might suffice. Reworded/rebased and force-pushed.

> Looks like your rebase re-added one of my latest commits
> 
> ![image](https://user-images.githubusercontent.com/25299219/280491090-e6f256ae-d315-4984-964d-4d746e170a2f.png)

:facepalm: par for the course of yesterday, thanks, fixed.

c-po committed:
T5713: only strip "secret" CLI node and nothing else

Commit 30eb308149 ("T5713: Strip string after "secret" in IPSEC config") had
good intention but this will happen:

use-secret foo CLI node will become " secret xxxxxx" so the output of
strip-private invalidates the configuration.

This has been changed to an exact match of "secret" only

c-po commented:
@MergifyIo backport sagitta
